### PR TITLE
Fix reading symlink file format

### PR DESCRIPTION
**Pull request by cberner**
Symlinks may cross HDFS instances. Previously we were using the current
filesystem for reading the sizes of the targets of the symlink. We now
always create a new filesystem for each symlink target, that matches its
path.

### CODE REVIEW

File: presto-hive/src/main/java/com/facebook/presto/hive/BackgroundHiveSplitLoader.java [hunk 1]
```diff
@@ -382,6 +394,31 @@ private void loadPartition(HivePartitionMetadata partition)
...
 
+    private static List<Path> getTargetPathsFromSymlink(Configuration conf, Path symlinkDir)
+    {
+        try {
+            FileSystem fileSystem = symlinkDir.getFileSystem(conf);
+            FileStatus[] symlinks = fileSystem.listStatus(symlinkDir, HIDDEN_FILES_PATH_FILTER);
+            List<Path> targets = new ArrayList<>();
+
+            for (FileStatus symlink : symlinks) {
+                try (BufferedReader reader = new BufferedReader(new InputStreamReader(fileSystem.open(symlink.getPath())))) {
```
Comment (martint) [inline presto-hive/src/main/java/com/facebook/presto/hive/BackgroundHiveSplitLoader.java]: What charset is supposed to be used for decoding these?

Comment (martint) [inline presto-hive/src/main/java/com/facebook/presto/hive/BackgroundHiveSplitLoader.java]: Guava has utilities for reading lines from a file or stream. Take a look at CharStreams.

Comment (cberner) [inline presto-hive/src/main/java/com/facebook/presto/hive/BackgroundHiveSplitLoader.java]: I just copied this from the Hadoop code, so this uses the same charset it previously did

Comment (martint) [inline presto-hive/src/main/java/com/facebook/presto/hive/BackgroundHiveSplitLoader.java]: Well, that code in hadoop is bad. It's sensitive to the system charset of the machine running the code.

Comment (cberner) [inline presto-hive/src/main/java/com/facebook/presto/hive/BackgroundHiveSplitLoader.java]: At least for their tests it looks like UTF8 is expected, so I'll assume that's the case here: https://issues.apache.org/jira/browse/HIVE-2811

**Comment by martint** [comment]
Looks good, but add the explanation we talked about offline.

**Pull request merged**